saml/src/main/java/com/linecorp/armeria/saml/JwtBasedSamlRequestIdManager.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2018 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.saml;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+import java.io.UnsupportedEncodingException;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.Date;
+
+import com.auth0.jwt.JWT;
+import com.auth0.jwt.JWTVerifier;
+import com.auth0.jwt.algorithms.Algorithm;
+import com.auth0.jwt.interfaces.DecodedJWT;
+import com.google.common.base.MoreObjects;
+
+import io.netty.util.internal.ThreadLocalRandom;
+
+/**
+ * A {@link SamlRequestIdManager} implementation based on JSON Web Tokens specification.
+ * See <a href="https://jwt.io/">JSON Web Tokens</a> for more information.
+ */
+final class JwtBasedSamlRequestIdManager implements SamlRequestIdManager {
+
+    private static final String CLAIM_NAME_SALT = "salt";
+
+    private final String issuer;
+    private final int validSeconds;
```

Comment:
Since we don't need to worry about allocation overhead, consider using `Duration`

Reply:
I wanted to use `Duration` but JWT library is based on `seconds`. So `Duration` may be confused to our user. It may look like that we can support `milliseconds` for expiration.